store/storage_manager_builder.go

```diff
@@ -32,48 +32,123 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 )
 
-// Builder centralizes dependency initialization.
+// StorageManagerBuilder centralizes dependency initialization.
 // It ensures proper typing and avoids redundant logic scattered across functions.
-type Builder struct {
+type StorageManagerBuilder struct {
 	ctx     context.Context
 	log     logging.Logger
 	metrics metrics.Metricer
 
-	memConfig *memconfig.SafeConfig
-
 	managerCfg     Config
+	memConfig      *memconfig.SafeConfig
 	v1VerifierCfg  verify.Config
 	v1EdaClientCfg clients.EigenDAClientConfig
 	v2ClientCfg    common.ClientConfigV2
 	v2SecretCfg    common.SecretConfigV2
+
+	putRetries                 uint
+	maxBlobSize                uint
+	eigenDaCertVerifierAddress string
 }
 
-func NewBuilder(
-	ctx context.Context, cfg Config,
+// NewStorageManagerBuilder creates a builder which knows how to build an IManager
+func NewStorageManagerBuilder(
+	ctx context.Context,
+	log logging.Logger,
+	metrics metrics.Metricer,
+	managerConfig Config,
 	v1VerifierCfg verify.Config,
 	v1EdaClientCfg clients.EigenDAClientConfig,
 	v2ClientCfg common.ClientConfigV2,
 	v2SecretCfg common.SecretConfigV2,
 	memConfig *memconfig.SafeConfig,
-	log logging.Logger,
-	metrics metrics.Metricer,
-) *Builder {
-	return &Builder{
+	putRetries uint,
+	maxBlobSize uint,
+	eigenDaCertVerifierAddress string,
+) *StorageManagerBuilder {
+	return &StorageManagerBuilder{
 		ctx,
 		log,
 		metrics,
+		managerConfig,
 		memConfig,
-		cfg,
 		v1VerifierCfg,
 		v1EdaClientCfg,
 		v2ClientCfg,
 		v2SecretCfg,
+		putRetries,
+		maxBlobSize,
+		eigenDaCertVerifierAddress,
+	}
+}
+
+// Build builds the storage manager object
+func (smb *StorageManagerBuilder) Build(ctx context.Context) (IManager, error) {
```

Comment:
Do we have to return an interface here?

Reply:
@samlaf probably not but I don't see any benefits to the latter